prettier.config.cjs

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// Shared prettier configuration for use in across the fluid-framework repository.
+// Individual packages may extend this and override rules as needed, though for consistent formatting, package-local
+// overrides should be avoided unless absolutely necessary.
+module.exports = {
```

Comment:
I would vote for taking a dependency on build-common and just extending that config here.

Here is an example of that pattern in fluid-framework: https://github.com/microsoft/FluidFramework/blob/main/common/lib/common-definitions/prettier.config.cjs